File: backend/schema/article.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from .gamme import GammeLigneOut


class ArticleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    nom: str
    description: str | None = None
    source: str | None = None
    chaine: str | None = None
    # Date d'enregistrement de la gamme (celle de sa derniere Gamme).
    date_creation: datetime | None = None


class ArticleListOut(BaseModel):
    total: int
    skip: int
    limit: int
    items: list[ArticleOut]


class ArticleDetailOut(ArticleOut):
    gamme_id: int | None = None
    lignes_gamme: list[GammeLigneOut] = []


class ArticleCreate(BaseModel):
    code: str
    nom: str
    description: str | None = None
    source: str | None = None


class ArticleUpdate(BaseModel):
    code: str | None = None
    nom: str | None = None
    description: str | None = None
    source: str | None = None

File: frontend/src/types/article.ts
export interface Article {
  id: number
  code: string
  nom: string
  description: string | null
  source: string | null
  chaine: string | null
  date_creation: string | null
}

export interface ArticleListe {
  total: number
  skip: number
  limit: number
  items: Article[]
}

export interface GammeLigne {
  id: number
  ordre: number
  operation_libelle: string
  temps_equilibre: number | null
  target: number | null
  machine: string | null
  operateur_nom: string | null
}

export interface ArticleDetail extends Article {
  gamme_id: number | null
  lignes_gamme: GammeLigne[]
}
